tests/jsunit/xml_test.js

```diff
@@ -121,3 +160,44 @@ function test_appendDomToWorkspace() {
     workspace.dispose();
   }
 }
+
+function test_blockToDom_fieldToDom_trivial() {
+  Blockly.defineBlocksWithJsonArray([{
+    'type': 'field_variable_test_block',
+    'message0': '%1',
+    'args0': [
+      {
+        'type': 'field_variable',
+        'name': 'VAR',
+        'variable': 'item'
+      }
+    ],
+  }]);
+  xmlTest_setUpWithMockBlocks()
+  workspace.createVariable('name1', 'type1', 'id1');
+  var block = new Blockly.Block(workspace, 'field_variable_test_block');
+  block.inputList[0].fieldRow[0].setValue('name1');
+  var resultFieldDom = Blockly.Xml.blockToDom(block).childNodes[0];
+  xmlTest_checkFieldDomValues(resultFieldDom, 'VAR', 'type1', 'id1', 'name1')
```

Comment:
I think you mean when createVariable doesn't specify a type or id. If not, let me know.
Done.